plugin/core/configurations.py

```diff
@@ -61,16 +62,31 @@ def update(self) -> None:
                 debug("applying .sublime-project override for", name)
             else:
                 overrides = {}
+            if name in self._disabled_for_session:
+                overrides["enabled"] = False
             self.all[name] = ClientConfig.from_config(config, overrides)
         for name, c in project_settings.items():
             debug("loading project-only configuration", name)
             self.all[name] = ClientConfig.from_dict(name, c)
         self._window.run_command("lsp_recheck_sessions")
 
     def enable_config(self, config_name: str) -> None:
-        enable_in_project(self._window, config_name)
+        if not self._reenable_disabled_for_session(config_name):
+            enable_in_project(self._window, config_name)
         self.update()
 
-    def disable_config(self, config_name: str) -> None:
-        disable_in_project(self._window, config_name)
+    def disable_config(self, config_name: str, only_for_session: bool = False) -> None:
+        if only_for_session:
+            self._disable_for_session(config_name)
+        else:
+            disable_in_project(self._window, config_name, only_for_session)
         self.update()
+
+    def _disable_for_session(self, config_name: str) -> None:
+        self._disabled_for_session.add(config_name)
+
+    def _reenable_disabled_for_session(self, config_name: str) -> bool:
+        if config_name in self._disabled_for_session:
+            self._disabled_for_session.remove(config_name)
+            return True
+        return False
```

Comment:
This does two tree traversals ;) you can do it with one tree traversal by just doing the `.remove` call and catching the KeyError

Reply:
Changed.